modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java

```diff
@@ -1656,8 +1656,14 @@ private String cacheInfo(GridCacheContext cacheCtx) {
 
                 int partId = e0.getKey();
 
-                assert cctx.wal().reserved(cpEntry.checkpointMark())
-                    : "WAL segment for checkpoint " + cpEntry + " has not reserved";
+                if (reservedCheckpointMark != null && !cctx.wal().reserved(reservedCheckpointMark)) {
+                    log.warning("Reservation failed because the segment was released: " + reservedCheckpointMark);
+
+                    reservedForExchange = null;
+
+                    grpPartsWithCnts.clear();
```

Comment:
Hmm... Do we need to clear `grpPartsWithCnts` map completely and just switch to the next cache group?

Reply:
I think we should, because if the oldest segment is released, then for all groups we no longer guarantee a historical rebalance.

https://github.com/apache/ignite/blob/74b215580b4ebb7d883025259e7e29502b8e1a3e/modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java#L1639

Reply:
Have personally discussed not to continue iterating over the groups.